state-sync/state-sync-v2/data-streaming-service/src/data_stream.rs

```diff
@@ -578,6 +575,13 @@ impl<T: DiemDataClient + Send + Clone + 'static> DataStream<T> {
         Ok(())
     }
 
+    /// Terminates the stream by aborting all spawned tasks
+    pub fn terminate(&mut self) {
+        for spawned_task in &self.spawned_tasks {
+            spawned_task.abort();
+        }
+    }
```

Comment:
I think this would work well in the destructor?

Reply:
Aah, nice!